File: Problems/637_Average_of_Levels_in_Binary_Tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<double> averageOfLevels(TreeNode* root) {
        vector<double> res;
        queue<TreeNode*> qu;
        qu.push(root);
        while (!qu.empty()) {
            int len = qu.size();
            double total = 0;
            for (int i = 0; i < len; ++i) {
                TreeNode* node = qu.front();
                qu.pop();
                total += node -> val;
                if (node -> left) {
                    qu.push(node -> left);
                }
                if (node -> right) {
                    qu.push(node -> right);
                }
            }
            res.push_back(total / len);
        }
        return res;
    }
};

File: Problems/637_Average_of_Levels_in_Binary_Tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def averageOfLevels(self, root: Optional[TreeNode]) -> List[float]:
        res = []
        qu = deque([root])
        
        while qu:
            length = len(qu)
            total = 0
            for _ in range(length):
                node = qu.popleft()
                total += node.val
                if node.left:
                    qu.append(node.left)
                if node.right:
                    qu.append(node.right)
            res.append(total / length)
        return res